ports/raspberrypi/bindings/rp2pio/StateMachine.c

```diff
@@ -618,23 +619,166 @@ const mp_obj_property_t rp2pio_statemachine_writing_obj = {
 };
 
 
+
 //|     pending: int
+//|     pending_write: int
 //|     """Returns the number of pending buffers for background writing.
 //|
 //|     If the number is 0, then a `StateMachine.background_write` call will not block."""
-static mp_obj_t rp2pio_statemachine_obj_get_pending(mp_obj_t self_in) {
+static mp_obj_t rp2pio_statemachine_obj_get_pending_write(mp_obj_t self_in) {
+    rp2pio_statemachine_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    return mp_obj_new_int(common_hal_rp2pio_statemachine_get_pending_write(self));
+}
+MP_DEFINE_CONST_FUN_OBJ_1(rp2pio_statemachine_get_pending_write_obj, rp2pio_statemachine_obj_get_pending_write);
+
+const mp_obj_property_t rp2pio_statemachine_pending_write_obj = {
```

Comment:
Please use the MP_PROPERTY_GETTER macro here, even though it wasn't used in the code whose style you were emultaing elsewhere in this file.  We should tidy away the direct uses of `const mp_obj_property_t` that have not previously been done, like the other ones in this file.